tests/system_tests/services_test_app.py

```diff
@@ -79,9 +84,32 @@ def get(self):
         assert memcache.get('test') == 'hi', 'Memcache failure'
 
 
+def socket_thread():
+    logging.info('In socket thread')
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
+    s.bind(('', 1000))
+    s.listen(5)
+    while True:
+        c, addr = s.accept()
+        data = c.recv(1024)
+        c.send(data)
+        c.close()
+
+
+class OpenPort(webapp2.RequestHandler):
+    """Open port 1000."""
```

Comment:
The "Open port 1000?"  It's intended for viewers of the web interface which is in the OpenPort class, so I want it there at minimum.  I'll add more detailed commentary to socket_thread(), though.